Gems/AtomLyIntegration/CommonFeatures/Code/Source/Material/EditorMaterialComponentInspector.cpp

```diff
@@ -192,7 +192,21 @@ namespace AZ
             {
                 if (!IsLoaded())
                 {
-                    m_overviewText->setText(tr("Selected entities and material slots must have valid and compatible materials."));
+                    if (m_entityIdsToEdit.size() > 1)
+                    {
+                        m_overviewText->setText(tr("The selected entities and materials cannot be edited.\n"
+                                                   "Multiple entities and materials have been selected for editing.\n"
+                                                   "All of the selected entities must be valid, active, and have a material component.\n"
+                                                   "Each material component must provide the selected material slot.\n"
+                                                   "The active material on each slot must have the same material type."));
+                    }
+                    else
+                    {
+                        m_overviewText->setText(tr("The selected entities and materials cannot be edited.\n"
+                                                   "The selected entity must be valid, active, and have a material component.\n"
+                                                   "The material component must provide the selected material slot."));
```

Comment:
This statement might be confusing, because there shouldn't be any way to get to the property inspector if the material component doesn't provide the selected material slot.

Reply:
One of these message blocks will only appear if something went wrong while opening the dialog or if one of the entities changed something after it was open.
One of the selected entities could be deleted. 
Mesh or material components could be removed.
One of the models or materials could be replaced, reloaded, or reassigned to something that doesn't have a matching slot or material type or is otherwise not compatible with the rest of the selection.

I tested different combinations of selecting multiple entities, modifying properties, undo, redo, changing or deleting only one of the entities in the set, opening multiple dialogs with different selections that had overlapping entities.